warehouse/malware/tasks.py

```diff
@@ -21,10 +23,30 @@
 
 
 @task(bind=True, ignore_result=True, acks_late=True, retry_backoff=True)
-def run_check(task, request, check_name, obj_id):
-    check = getattr(checks, check_name)(request.db)
+def run_check(task, request, check_name, obj_id=None, manually_triggered=False):
+    try:
+        check = getattr(checks, check_name)(request.db)
+    except NoResultFound:
+        request.log.info("Check %s isn't active. Aborting." % check_name)
```

Comment:
Is it the case that it isn't _active_ here or that it doesn't exist in the codebase? This might be a bit misleading.

Reply:
If it does not exist in the code base, an `AttributeError` will be raised, which isn't caught here.

Reply:
> Is it the case that it isn't active here or that it doesn't exist in the codebase?
It is the case that it isn't active (e.g. in the `enabled` or `evaluation` state). Loading info from the database happens when the check class is initialized. That is why I'm only catching the `NoResultFound` exception.

Reply:
Ah! I misinterpreted NoResultFound with AttributeError in my mind :)